### PR TITLE
DEVEX-2420 Bahrain release prep

### PR DESCRIPTION
**Pull request by aogorodnikov-dna**
Adding configs for Bahrain release

### CODE REVIEW

File: scripts/docker_image/Dockerfile [hunk 1]
```diff
@@ -4,4 +4,8 @@ ARG BASE_DIR
 
 COPY ${BASE_DIR}/dxCompiler-$VERSION.jar /dxCompiler.jar
 
-ENTRYPOINT ["java", "-jar", "/dxCompiler.jar"]
```
Comment (r-i-v-a) [inline scripts/docker_image/Dockerfile]: Only add the warning script to the legacy Docker image (that's published as part of the release GHA). The other Dockerfile (I think it's this one in scripts/docker_image) is part of the instructions to DIY it, which is the new way.

File: scripts/docker_image/docker_image_entry_test.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+echo "WARNING: This Docker image will be removed permanently on 2024/12/1. Please STOP using this image ASAP"
```
Comment (r-i-v-a) [inline scripts/docker_image/docker_image_entry_test.sh]: Move to be with legacy Dockerfile

File: scripts/dxcompiler_copy/dxapp.json [hunk 3]
```diff
@@ -109,7 +116,7 @@
     }
   },
   "details": {
-    "whatsNew": "* 0.0.5: updated azure instances to v2 for OFH region\n * 0.0.3: Added new London OFH region\n * 0.0.2: Added new London region."
+    "whatsNew": "* 0.0.6: Added new Bahrain AWS region aws:me-south-1 \n * 0.0.5: updated azure instances to v2 for OFH region\n * 0.0.3: Added new London OFH region\n * 0.0.2: Added new London region."
```
Comment (r-i-v-a) [inline scripts/dxcompiler_copy/dxapp.json]: minor: extra space

File: scripts/multi_region_tests.py [hunk 4]
```diff
@@ -28,7 +28,8 @@
             "dxCompiler_Berlin",
             "dxCompiler_London",
             "dxCompiler_Europe_London",
-            "dxCompiler_OFH_TRE_London"]
+            "dxCompiler_OFH_TRE_London",
+            "dxCompiler_Bahrain"]
```
Comment (r-i-v-a) [inline scripts/multi_region_tests.py:32]: Thanks for updating the multiregion tests

**Review by r-i-v-a — CHANGES_REQUESTED** [review]
Only add the warning label to the legacy Docker image (the one built as part of release GHA). 

What is described here https://github.com/dnanexus/dxCompiler/blob/develop/scripts/docker_image/README.md is the new way and doesn't need the warning.

File: scripts/docker_image/Dockerfile (hunk 1 above)
Comment (aogorodnikov-dna) [reply scripts/docker_image/Dockerfile]: I see, I just used it for tests.

Comment (aogorodnikov-dna) [reply scripts/docker_image/Dockerfile]: fixed in [439e46e](https://github.com/dnanexus/dxCompiler/pull/481/commits/439e46e14198b41b361e2ed325662acef845179c)

File: scripts/docker_image/docker_image_entry_test.sh (hunk 2 above)
Comment (aogorodnikov-dna) [reply scripts/docker_image/docker_image_entry_test.sh]: not moved, just needed to delete from here. Done in [439e46e](https://github.com/dnanexus/dxCompiler/pull/481/commits/439e46e14198b41b361e2ed325662acef845179c)

File: scripts/dxcompiler_copy/dxapp.json (hunk 3 above)
Comment (aogorodnikov-dna) [reply scripts/dxcompiler_copy/dxapp.json]: hopefully in [439e46e](https://github.com/dnanexus/dxCompiler/pull/481/commits/439e46e14198b41b361e2ed325662acef845179c)

File: scripts/Dockerfile [hunk 5]
```diff
@@ -11,8 +11,4 @@ RUN apt-get update && \
 # dxCompiler
 COPY dxCompiler-$VERSION.jar /dxCompiler.jar
 
-# Copy the entrypoint script
```
Comment (r-i-v-a) [inline scripts/Dockerfile]: The entrypoint script should still be kept for this one, though

**Review by r-i-v-a — CHANGES_REQUESTED** [review]
Keep entrypoint script for this Dockerfile: scripts/Dockerfile

File: scripts/Dockerfile (hunk 5 above)
Comment (aogorodnikov-dna) [reply scripts/Dockerfile]: sorry, just mixed up when reverting.

**Review by r-i-v-a — APPROVED** [review]

**Pull request merged**